invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -483,30 +485,67 @@ class TileResamplerProcessorInvocation(ImageProcessorInvocation):
 
     # res: int = InputField(default=512, ge=0, le=1024, description="The pixel resolution for each tile")
     down_sampling_rate: float = InputField(default=1.0, ge=1.0, le=8.0, description="Down sampling rate")
+    mode: Literal["regular", "blur", "var", "super"] = InputField(
+        default="regular", description="The controlnet tile model being used"
+    )
+
+    def apply_gaussian_blur(self, image_np: np.ndarray[Any, Any], ksize: int = 5, sigmaX: float = 1.0):
+        if ksize % 2 == 0:
+            ksize += 1  # ksize must be odd
+        blurred_image = cv2.GaussianBlur(image_np, (ksize, ksize), sigmaX=sigmaX)
+        return blurred_image
+
+    def apply_guided_filter(self, image_np: np.ndarray[Any, Any], radius: int, eps: float, scale: int):
+        filter = FastGuidedFilter(image_np, radius, eps, scale)
+        return filter.filter(image_np)
+
+    # based off https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic
+    def tile_resample(self, np_img: np.ndarray[Any, Any]):
```

Comment:
As written, it is hard to understand the interactions between the various modes in `tile_resample(...)`. How about we update `run_processor(...)` to look like this:

```python
    def run_processor(self, image: Image.Image) -> Image.Image:
        np_img = np.array(image, dtype=np.uint8)
        if self.mode == "regular":
            processed_np_image = self._regular_resample(np_img)
        elif self.mode == "blur":
            processed_np_image = self._blur_resample(np_img)
        elif self.mode == "var":
            processed_np_image = self._var_resample(np_img)
        elif self.mode == "super":
            processed_np_image = self._super_resample(np_img)
        else:
            raise ValueError(f"Invalid mode: {self.mode}")
        processed_image = Image.fromarray(processed_np_image)
        return processed_image
```

Reply:
Made these changes.